gtsfm/frontend/verifier/oanet.py

```diff
@@ -0,0 +1,188 @@
+"""Wrapper around Order Aware Net verifier from Zhang et al.
+
+See: https://arxiv.org/abs/1908.04964
+Wraps the MIT-licensed code from https://github.com/zjhthu/OANet.
+
+Authors: Ayush Baid, John Lambert
+"""
+from pathlib import Path
+from types import SimpleNamespace
+from typing import Optional, Tuple
+
+import numpy as np
+import torch
+from gtsam import Cal3Bundler, Rot3, Unit3
+
+import gtsfm.frontend.verifier.verifier_base as verifier_base
+import gtsfm.utils.logger as logger_utils
+from gtsfm.common.keypoints import Keypoints
+from gtsfm.frontend.verifier.ransac import Ransac
+from gtsfm.frontend.verifier.verifier_base import VerifierBase
+from thirdparty.OANet.core.oan import OANet
+
+logger = logger_utils.get_logger()
+
+OANET_MODEL_FPATH = (
+    Path(__file__).resolve().parent.parent.parent.parent
+    / "thirdparty"
+    / "OANet"
+    / "weights"
+    / "gl3d"
+    / "sift-4000"
+    / "model_best.pth"
+)
+
+
+class OANetVerifier(VerifierBase):
+    def __init__(
+        self,
+        use_intrinsics_in_verification: bool,
+        estimation_threshold_px: float,
+    ) -> None:
+        """Initializes the verifier.
+
+        Due to our implementation structure, providing the descriptors to the verifier is not possible
+        (thus ratio test/mutual info unavailable).
+
+        Args:
+            use_intrinsics_in_verification: Flag to perform keypoint normalization and compute the essential matrix
+                instead of fundamental matrix. This should be preferred when the exact intrinsics are known as opposed
+                to approximating them from exif data.
+            estimation_threshold_px: maximum distance (in pixels) to consider a match an inlier, under squared
+                Sampson distance.
+        """
+        self._use_intrinsics_in_verification = use_intrinsics_in_verification
+        self._estimation_threshold_px = estimation_threshold_px
+        self._min_matches = (
+            verifier_base.NUM_MATCHES_REQ_E_MATRIX
+            if self._use_intrinsics_in_verification
+            else verifier_base.NUM_MATCHES_REQ_F_MATRIX
+        )
+
+        # for failure, i2Ri1 = None, and i2Ui1 = None, and no verified correspondences, and inlier_ratio_est_model = 0
+        self._failure_result = (None, None, np.array([], dtype=np.uint64), 0.0)
+
+        self.is_cuda = torch.cuda.is_available()
```

Comment:
Nit: all class attributes here and below can be private (with the underscore suffix)